### PR TITLE
Close api

### PR DESCRIPTION
**Pull request by dgageot**
This makes the libmachine Api store/close the opened plugins, instead of relying on a global variable.

### CODE REVIEW

**Comment by jeanlaurent** [comment]
I took the time to read it all. LGTM

Code change is quite complex. I think it's worth that @nathanleclaire gives is voice on it.

File: commands/commands.go [hunk 1]
```diff
@@ -98,7 +97,8 @@ func runAction(actionName string, c CommandLine, api libmachine.API) error {
 
 func fatalOnError(command func(commandLine CommandLine, api libmachine.API) error) func(context *cli.Context) {
 	return func(context *cli.Context) {
-		api := libmachine.NewClient(mcndirs.GetBaseDir())
+		api := libmachine.NewClient(mcndirs.GetBaseDir(), mcndirs.GetMachineCertDir())
```
Comment (nathanleclaire) [inline commands/commands.go]: Why did the dir passed in here change?

**Comment by nathanleclaire** [comment]
Nice. :+1: 

Any ideas how to unit test this to ensure that all plugin servers have indeed been shut down?

LGTM

**Comment by nathanleclaire** [comment]
BTW running with `-race` this PR doesn't report any races on `create`, which is good news.

File: commands/commands.go (hunk 1 above)
Comment (dgageot) [inline commands/commands.go]: Yes. To be honest, it has to be improved in a near future. I'm thinking about how we could pass a `MachinesDirs` struct that would be used in different places.

**Comment by nathanleclaire** [comment]
Hey @dgageot, I'm confused about what happened in this PR.  It says that you merged in one commit above, but there are 4 commits in this PR, and it is still open.

Should this be rebased and/or merged?

**Comment by dgageot** [comment]
I have no idea. I see all 4 commits merged in master and the PR is marked as closed for me...

**Comment by nathanleclaire** [comment]
OK maybe Github hiccup or me mis-reading things.  It looks good now, thanks.

**Pull request merged**